### PR TITLE
ipynb to mdx

### LINKED ISSUE #1346: Tutorials documentation (sidebar)
### Issue Description
Update the tutorials section in the documentation for better UX when clicking through the tutorials and the sidebar.

### PR DESCRIPTION
**Pull request by ndmlny-qs**
This commit contains the following changes.

- Updates to the tutorials plotting module. Changes here were propagated
  to the tutorials using them. They include `baseball.py` and `nba.py`.
- Tutorials were modified so they can accommodate conversion to mdx.
- The following tutorials have been updated to include interactive
  figures:
  - `Bayesian_Logistic_Regression`
  - `Robust_Linear_Regression`
  - `Sparse_Logistic_Regression`
- The `website/tutorials.json` file was updated to reflect the tutorials
  that can be successfully transformed into mdx files.

Resolves #1346

### Types of changes
- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](https://github.com/facebookresearch/beanmachine/blob/main/CONTRIBUTING.md)** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

# Bugs
- General
  - Remove `tqdm` progress bars in the mdx output
- Tutorials
  - Coin flipping
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: matplotlib
      - :x: plotly
    - Create a ticket to update with interactive figures.
  - Linear regression
    - Contains errors, create a ticket.
  - Robust linear regression tutorial
    - :x: runs in Colab (needs updated `plots.py` module)
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
  - Logistic regression tutorial
    - :x: runs in Colab (needs updated `plots.py` module)
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
  - Sparse logistic regression tutorial
    - :x: runs in Colab (needs updated `plots.py` module)
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
  - Hierarchical regression
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
    - Create a ticket to ensure all figures are interactive.
  - Hierarchical modeling
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
    - Create a ticket to ensure all figures are interactive.
  - IRT tutorial
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
    - Create ticket to update displays of Markdown dataframes.
  - Zero inflated count data
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: Bokeh
    - Create a ticket to ensure all figures are interactive.
  - Hidden Markov model
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: matplotlib
    - Create a ticket to update with interactive figures.
  - Gaussian mixture model
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :x: Plotly
    - Create a ticket to update with interactive figures.
  - Neal's funnel
    - :heavy_check_mark: runs in Colab
    - plots visible before execution in Colab
      - :heavy_check_mark: matplotlib
      - :x: plotly
    - Create a ticket to update with interactive figures.
  - Gaussian process using gpytorch
    - Contains errors, create a ticket.

### CODE REVIEW

**Comment by ndmlny-qs** [comment]
This is a large PR, with many changes to the tutorials for inclusion in the website.

**Comment by ndmlny-qs** [comment]
@jpchen someone with privileges to change the GitHub actions will need to add `export NODE_OPTIONS=--max-old-space-size=8192` when building the documentation. Some of the tutorials are quite large due to the resolution needed for allowing figures to show in Colab, which was to add the  `output_notebook()` command to many cells.

**Comment by jpchen** [comment]
Thanks for doing this! The import tool seems to be broken so in the meantime I will review via reviewnb and make the changes for the internal tests when the import tool works.

File: tutorials/Bayesian_Logistic_Regression.ipynb [hunk 1]
```diff
@@ -3,13 +3,30 @@
   {
```
Comment (zaxtax) [inline tutorials/Bayesian_Logistic_Regression.ipynb:3]: <p>This and subsequent plots do not display in colab.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/facebookresearch/beanmachine/pull/1392/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

File: src/beanmachine/tutorials/utils/plots.py [hunk 2]
```diff
@@ -2,43 +2,85 @@
 #
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
-
```
Comment (horizon-blue) [inline src/beanmachine/tutorials/utils/plots.py:5]: Our internal linter is complaining about the lack of a new line after the comments at the beginning of the file (before the docstring) :)

**Comment by horizon-blue** [comment]
Thanks for creating the detailed summary for what's going on on each of the tutorials, Andy! I just tried out your script and it works wonderfully on machine :D.

The only thing that I'm not too sure about is the progress bar: I think we can just leave them in the output instead of removing them ([just like how other PPLs renders their tutorials](https://docs.pymc.io/en/v3/pymc-examples/examples/pymc3_howto/api_quickstart.html)). It seems like we skips the progress bar by [skipping the `stderr` cells](https://github.com/facebookresearch/beanmachine/blob/aed1e59b1e677d2fd3537ede14a6da2eb8f89780/scripts/convert_ipynb_to_mdx.py#L333-L334). I think we can just merge the texts from both `stdout` and `stderr` into a single output cell (just like how Jupyter renders the output); otherwise, this can happen:

4 chain progress bar becomes 4 output cell
(i.e 4 progress bars cause the output to be split into 4 cells)

Other than this, it's a great work overall! Because this PR touches many `ipynb` files, we should aim to merge this PR as soon as possible to avoid blocking other works or introduce merge conflicts. We can fix the few problematic tutorials in follow up PRs

**Comment by ndmlny-qs** [comment]
@horizon-blue I don't have a pretty work-around for this. When we read in the notebook, for some reason we lose `tqdm` information. So, I can get the tutorials to output something like below,

tqdm-tutorial

but that may be less useful than just not showing the progress bars in the rendered tutorials. I will push the version where we do not keep the progress bars around, but it does have a flag to render what is in the above image if we so wish.

**Comment by horizon-blue** [comment]
> When we read in the notebook, for some reason we lose tqdm information.

I see. To me it's still fine to keep the progress bar around just to hint users that "something is going on here," though I will let you make the decision here :). Thanks for addressing this so quickly

One thing I noticed is that with the last commit you made, outputs from `%%time` and `BMGInference` seems to be stripped as well, e.g. the end of [Coin flipping tutorial](https://github.com/facebookresearch/beanmachine/blob/aed1e59b1e677d2fd3537ede14a6da2eb8f89780/tutorials/Coin_flipping.ipynb) contains a session that runs BMG and measures its time, but the output is removed when converting to mdx file. Any idea what's going on here?

**Comment by ndmlny-qs** [comment]
> outputs from `%%time` and `BMGInference` seems to be stripped as well

that's a regression, fixing now

**Review by horizon-blue — APPROVED** [review]
With the latest changes, this PR looks great to me now. This will be a great addition to our website. Thanks, Andy!

I'm going to re-import it and merge the PR when the tests finish. As mentioned earlier, we can work on the rest of tutorials in a separate PR when the related issues are fixed :).

**Comment by jpchen** [comment]
This looks great! One nit is that I think we can remove the ['tutorials landing page'](https://beanmachine.org/docs/tutorials/) now since it is a bit redundant.

**Pull request closed without merging**